### PR TITLE
coq-of-ocaml: add version 2.5.3

### PR DESCRIPTION
**Pull request by clarus**
There are versions for OCaml 4.12, 4.13, and 4.14. Thanks!

### CODE REVIEW

File: packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.12/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+build: [
+  ["sh" "-c" "cd proofs && ./configure.sh"] {coq:installed}
+  [make "-C" "proofs" "-j%{jobs}%"] {coq:installed}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+install: [
+  [make "-C" "proofs" "install"] {coq:installed}
+]
+depends: [
+  "angstrom"
```
Comment (raphael-proust) [inline packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.12/opam:17]: You are missing a lower-bound for this constraint. This is causing the CI error:

```
# File "src/constant.ml", line 55, characters 13-25:
# 55 | let dquote = Angstrom.map ~f:(fun _ -> PDQuote) (Angstrom.char '"')
#                   ^^^^^^^^^^^^
# Error: Unbound value Angstrom.map
```

I think the `map` function was introduced in 0.15.0: https://github.com/inhabitedtype/angstrom/commit/93f29c2343491ef773efc89d2fa4e695a5bd1202

So I'd suggest

```suggestion
  "angstrom" {>= "0.15.0"}
```

If this is correct, then you should also upstream this change.

Comment (raphael-proust) [reply packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.12/opam:17]: Note: the same change is needed on all three package files.

Comment (clarus) [reply packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.12/opam]: Thanks! I have added the constraint for all three packages.

File: packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.14/opam [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  ["sh" "-c" "cd proofs && ./configure.sh"] {coq:installed}
+  [make "-C" "proofs" "-j%{jobs}%"] {coq:installed}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+install: [
+  [make "-C" "proofs" "install"] {coq:installed}
+]
+depends: [
+  "angstrom" {>= "0.15.0"}
+  "csexp"
```
Comment (raphael-proust) [inline packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.14/opam:18]: There seems to be a lower-bound constraint missing here:
[lower-bounds (failed: Unbound type constructor Csexp.t)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/5114feaf36e3735c6a5def87485d58a854d4077e/variant/compilers,4.14,coq-of-ocaml.2.5.3+4.14,lower-bounds)

I'm not sure what exact version you need, but the downgrade which reveals the error is:
```
downgrade csexp              1.5.1 to 1.0.0      [required by coq-of-ocaml]
```

Comment (clarus) [reply packages/coq-of-ocaml/coq-of-ocaml.2.5.3+4.14/opam]: OK, thanks! I added a lower bound constraint.

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**